### PR TITLE
Decomposed Training Loop

### PR DESCRIPTION
**Pull request by rdiehlmartinez**
First stab at decomposed, configurable training loop

* Configuration files broken down into a hierarchical structure
* Configuration files contain 4 main configurable packages: model, dataset, tokenizer, objective 

* Huggingface dataset model specified for custom data loading 

* Tokenizer automatically retrains on the baby lm dataset if it originates from an org outside ours; tuned tokenizer then pushed up to huggingface hub 
* Model cls includes a registry and sets up skeleton for overwriting base models 

TODO:
* Decompose the objective function into a class that returns a collator as well as method for computing the loss

### CODE REVIEW

**Review by codebyzeb — APPROVED** [review]
Fixed requirements, otherwise looks good
